celest/coordinates/astronomical_quantities.py

```diff
@@ -0,0 +1,602 @@
+    else:
+        jd = np.array(julian.to(u.jd2000)) + 0.5
+    a, f = jd.astype(int), jd % 1
+
+    ind = np.where(a >= 2291161)[0]
+    alpha = ((a[ind] - 1867216.25) / 36524.25).astype(int)
+    a[ind] = a + 1 + alpha - (alpha / 4).astype(int)
+
+    b = a + 1524
+    c = ((b - 122.1) / 365.25).astype(int)
+    d = (365.25 * c).astype(int)
+    e = ((b - d) / 30.6001).astype(int)
+
+    day = b - d - (30.6001 * e).astype(int) + f
+
+    month = e
+    idx1 = np.where(month < 14)[0]
+    idx2 = np.where((month == 14) | (month == 15))[0]
+    month[idx1] = month[idx1] - 1
+    month[idx2] = month[idx2] - 13
+
+    year = c
+    idx1 = np.where(month > 2)[0]
+    idx2 = np.where((month == 1) | (month == 2))[0]
+    year[idx1] = year[idx1] - 4716
+    year[idx2] = year[idx2] - 4715
+
+    return year, month, day
+
+
+def day_of_year(julian: Quantity) -> np.ndarray:
+    """Return the day of the year.
+
+    Parameters
+    ----------
+    julian : Quantity
+        Quantity object containing date data in the JD2000 epoch.
+
+    Returns
+    -------
+    np.ndarray
+        1-D array containing the day of the year.
+
+    Notes
+    -----
+    The day of year is calculated using the methods in "Astronomical
+    Algorithms" by Jean Meeus. [Mee98g]_
+
+    References
+    ----------
+    .. [Mee98g] Jean Meeus. Astronomical algorithms. 2nd ed. Willmann-Bell,
+       1998, pp. 65. isbn: 9780943396613.
+
+    Examples
+    --------
+    >>> julData = Quantity(2447273.5, u.jd2000)
+    >>> day_of_year(julian=julian)
+    113
+    """
+    year, month, day = from_julian(julian)
+
+    K = np.full(day.shape, 2)
+    K[np.where(year % 4 == 0)] = 1
+
+    N = (275 * month / 9).astype(int)
+    N = N - K * ((month + 9) / 12).astype(int) + day - 30
+
+    return N.astype(int)
+
+
+def equation_of_time(julian: Quantity) -> Quantity:
+    """Return the equation of time in decimal degrees.
+
+    Parameters
+    ----------
+    julian : Quantity
+        Quantity object containing date data in the JD2000 epoch.
+
+    Returns
+    -------
+    Quantity
+        Quantity object containing the Equation of Time in decimal degrees.
+
+    Notes
+    -----
+    The equation of time is calculated using the methods described in
+    "Astronomical Algorithms" by Jean Meeus. [Mee98h]_
+
+    References
+    ----------
+    .. [Mee98h] Jean Meeus. Astronomical algorithms. 2nd ed. Willmann-Bell,
+       1998, pp. 184 - 185. isbn: 9780943396613.
+
+    Examples
+    --------
+    >>> julian = Quantity(2448908.5, u.jd2000)
+    >>> equation_of_time(julian=julian)
+    3.427351
+    """
+    sun_mean_longitude = get_mean_longitude_of_sun(julian).to(u.rad)
+    sun_mean_anomaly = get_mean_anomaly_of_sun(julian).to(u.rad)
+    earth_eccentricity = get_earth_eccentricity(julian).to(u.rad)
+
+    sun_apparent_obliquity = apparent_obliquity(julian).to(u.rad)
+    y = np.tan(sun_apparent_obliquity / 2) ** 2
+
+    time_equation = y * np.sin(2 * sun_mean_longitude) - \
```

Comment:
Would it be cleaner to break this into multiple variables and add them together like earlier functions?